File: Ejemplos/41-TenistasProjectReactor/src/main/java/dev/joseluisgs/models/Tenista.java
package dev.joseluisgs.models;

import java.time.LocalDate;

public record Tenista(
        Long id,
        String nombre,
        String pais,
        int altura,
        int peso,
        int puntos,
        Mano mano,
        LocalDate fechaNacimiento
) {
    public static Long NEW_ID = 0L;
}

File: Ejemplos/39-TenistasFlows/src/main/kotlin/models/Tenista.kt
package dev.joseluisgs.models

import java.time.LocalDate

data class Tenista(
    val id: Long = NEW_ID,
    val nombre: String,
    val pais: String,
    val altura: Int,
    val peso: Int,
    val puntos: Int,
    val mano: Mano,
    val fechaNacimiento: LocalDate,
) {

    companion object {
        const val NEW_ID = 0L
    }

    enum class Mano {
        DIESTRO,
        ZURDO
    }

}